src/main/java/no/hyper/imagecrop/ImageTray.java

```diff
@@ -0,0 +1,205 @@
+        int y = Math.round(cropRect.top) - Math.round(top);
+
+        Log.d("CROP", "rect: " + Math.round(cropRect.left) + ", left: " + Math.round(left));
+        Bitmap newBitmap = Bitmap.createBitmap(picture, (x > 0) ? x : 0, (y > 0) ? y : 0,
+                Math.round(cropSize / mScaleFactor), Math.round(cropSize / mScaleFactor));
+
+        return Bitmap.createScaledBitmap(newBitmap, Math.round(cropSize),
+                Math.round(cropSize), true);
+    }
+
+    private final ScaleGestureDetector.OnScaleGestureListener mScaleGestureListener
+            = new ScaleGestureDetector.SimpleOnScaleGestureListener() {
+
+        @Override
+        public boolean onScaleBegin(ScaleGestureDetector detector) {
+            return super.onScaleBegin(detector);
+        }
+
+        @Override
+        public boolean onScale(ScaleGestureDetector detector) {
+            float scale = detector.getScaleFactor();
+            float fx = detector.getFocusX();
+            float fy = detector.getFocusY();
+            float sw = picture.getWidth()*scale;
+            float sh = picture.getHeight()*scale;
+            float dx = sw - picture.getWidth();
+            float dy = sh - picture.getHeight();
+
+            if(picture.getWidth()*mScaleFactor*scale < cropSize ||
+                    picture.getHeight()*mScaleFactor*scale < cropSize) {
+                return true;
+            }
+
+            mScaleFactor *= scale;
+            int[] dimens = getCropSquareDimens();
+            cropRect.set(dimens[0], dimens[1], dimens[2], dimens[3]);
+
+            left = left - (fx / (picture.getWidth()*mScaleFactor))*dx;
+            top = top - (fy / (picture.getHeight()*mScaleFactor))*dy;
+
+            if(left >= cropRect.left) {
+                left = cropRect.left;
+            }
+
+            if(top >= cropRect.top) {
+                top = cropRect.top;
+            }
+
+            if(left + picture.getWidth() <= cropRect.right) {
+                left += cropRect.right - (left + picture.getWidth());
+            }
+
+            if(top + picture.getHeight() <= cropRect.bottom) {
+                top += cropRect.bottom - (top + picture.getHeight());
+            }
+
+            invalidate();
+
+            return true;
+        }
+        
+    };
+
+    private final GestureDetector.SimpleOnGestureListener mGestureListener
+            = new GestureDetector.SimpleOnGestureListener() {
+        @Override
+        public boolean onDown(MotionEvent e) {
+            return true;
+        }
+
+        @Override
+        public boolean onScroll(MotionEvent e1, MotionEvent e2, float distanceX, float distanceY) {
+            if(!mScaleDetector.isInProgress()) {
+
+                float newX = left - distanceX/mScaleFactor;
+                float newY = top - distanceY/mScaleFactor;
+
+                float minX = (middle.x + cropSize /2)/mScaleFactor
+                        - picture.getWidth();
```

Comment:
This can be one-lined without exceeding the column count limit, can't it? :smile: